dev-util/cmake/cmake-3.27.7.ebuild

```diff
@@ -60,7 +60,7 @@ S="${WORKDIR}/${MY_P}"
 
 LICENSE="BSD"
 SLOT="0"
-IUSE="${CMAKE_DOCS_USEFLAG} dap emacs gui ncurses qt6 test"
+IUSE="${CMAKE_DOCS_USEFLAG} bootstrap dap emacs gui ncurses qt6 test"
```

Comment:
I suspect it should actually be `+bootstrap`. It's not nice to vary behaviour based on installed packages and it's just an optimisation anyway, so we should default to doing the sensible thing (re-bootstrapping without being affected by the installed `cmake`).

Reply:
Are there any pros to skipping the bootstrap stage and just using the already installed cmake? If there aren't any significant ones it might be better to just always bootstrap it.

Reply:
I think it makes building CMake a bit quicker. Could you do some benchmarks?

Reply:
Time with bootstrap:
```
real	16m29.647s
user	97m20.136s
sys	3m52.050s
```
Time without bootstrap:
```
real	9m31.653s
user	53m45.292s
sys	2m19.462s
```

Reply:
That's a way bigger difference than I was expecting. Can you try it with hyperfine?

Reply:
I'll give that a try. Just for clarity, I'm currently putting this laptop through the wringer with doing like 4 big things at the same time, so times might be skewed a bit

Reply:
yeah please do it when the machine is idle...

Reply:
Gotcha, it might be a while until it's idle as GCC 14's *still* running tests

Reply:
hyperfine results:

```
Benchmark 1: USE=bootstrap emerge -1 cmake::local
  Time (mean ± σ):     310.137 s ± 11.217 s    [User: 2538.862 s, System: 115.428 s]
  Range (min … max):   296.942 s … 326.364 s    10 runs

Benchmark 2: USE=-bootstrap emerge -1 cmake::local
  Time (mean ± σ):     220.210 s ±  3.032 s    [User: 1743.114 s, System: 82.521 s]
  Range (min … max):   215.919 s … 226.892 s    10 runs

Summary
  'USE=-bootstrap emerge -1 cmake::local' ran
    1.41 ± 0.05 times faster than 'USE=bootstrap emerge -1 cmake::local'
```

Reply:
Personally, I would prefer not to enable "bootstrap" by default. If we have a working cmake, we should just use it.

Reply:
Good point, maybe if it fails while trying to build with itself, point the user to try rebuilding with `USE=bootstrap` as well?